### DISCUSSION TITLE
Enum csla property on change not calling rule in Csla 8

### DISCUSSION THREAD

**Discussion by JacoJordaan** [MarimerLLC/csla#3779] [Questions]
I have this enum property in my business class:
````
        public static readonly PropertyInfo<MailTemplateTypeOptions> ObjectTypeProperty = RegisterProperty<MailTemplateTypeOptions>(c => c.ObjectType);
        public MailTemplateTypeOptions ObjectType
        {
            get { return GetProperty(ObjectTypeProperty); }
            set { SetProperty(ObjectTypeProperty, value); }
        }
````
I have a business rule based on this property.
Calling checkrules is calling the rule, but a change in the value of the property is not calling the rule anymore.

This is working in Csla 5, but not in Csla 8 (latest preview version).
I remember my test cases were running fine in Csla 6 and 7, but I have not tried to go back to these versions to check again.

Is there any rule changes in Csla 8 that may cause this to happen?

**Comment by hurcane** [comment]
Does the business rule have MailTemplateTypeOptions as the primary property or is it an input property? This is likely to be strongly correlated with #3720.
